File: test/guessing_game.py
import random
import time

number = random.randint(0, 50)
guess = -1
trycount = 8
isNum = False
win = False
input = " "

def is_Number(s):
    try:
        int(s)
        return True
    except ValueError:
        return False

print "Guess a Number! Any Number! \n"
print "(from 0 to 50)"
while(guess != number and trycount > 0):
    input = raw_input("guess: ")
    while(isNum == False):
        if(not is_Number(input)):
            print "enter a number pls"
            input = raw_input("guess: ")
            if (is_Number(input)):
                break
        else:
            isNum = True

    guess = int(input)

    if(guess < number):
        print "Too low" 

    if (guess > number):
        print "Too high"

    trycount = trycount - 1

    if(trycount > 1):
        print "You have ", trycount, " guesses left!!" 
    elif(trycount == 1):
        print "You have ", trycount, " guess left!!"

if(guess==number):
    print "You guessed the number" 
    win = True
if (trycount == 0):
    print "You ran out of guesses :( "  

if(not win):
    print "The number was ", number, "! " 
    print "game over"  
else:
    print "gg"  
    
key = raw_input("press Q to quit \n")  

while(key != "q" and key != "Q"):
    key = raw_input()

File: test/guessing_game.cpp
#include <iostream>
#include <string>
#include <stdlib.h>
#include <ctype.h>
#include <time.h>  
using namespace std;

int main()
{
    srand (time(NULL));
    int number = rand() % 100 + 1; 
    int guess = -1;
    int trycount = 8;
    bool isNumber = false;
    bool win = false;
    string input = "";

    cout << "Guess a Number! Any Number!" << endl;
    while(guess != number && trycount > 0){
        cin >> input;
        while(isNumber == false){
            if(input.length() == 1){
                if(!isdigit(input[0])){
                        cout << "please enter a number: ";
                        cin >> input;
                }else{
                    isNumber = true;
                }
            }
            else if(input.length() > 1){
                int check = input.length();
                for(int i = 0; i < input.length(); i++){
                    if(!isdigit(input[i])){
                        check--;
                        cout << "please enter a number: ";
                        cin >> input;
                    }
                }
                if(check == input.length())
                    //cout << check <<" z ";
                    isNumber = true;
            }
        }
        int in = atoi(input.c_str());
        guess = in; 

        if(guess < number)
            cout << "Too low" << endl;

        if (guess > number)
            cout << "Too high" << endl;

        trycount--;

        if(trycount > 1)
            cout << "You have " << trycount << " guesses left!!" << endl;
        else if(trycount == 1)
            cout << "You have " << trycount << " guess left!!" << endl;
    }

    if(guess==number){
        cout << "You guessed the number" << endl;
        win = true;
    }
    if (trycount == 0)
        cout << "You ran out of guesses :( " << endl;

    if(!win)
        cout << "The number was " << number << "! " << endl << "game over" << endl;
    else
        cout << "gg" << endl;
        
    cout << "press Q to quit" << endl;
    string quit = "";
    while(quit != "q" && quit != "Q")
        cin >> quit;
    0;
}
